File: SantaFinder.Web/Models/NewOrder/NewOrder.cs
﻿using System;
using System.Collections.Generic;
using SantaFinder.Web.Models.Shared;

namespace SantaFinder.Web.Models
{
    public class NewOrder
    {
        public NewOrderAddress Address { get; set; }
        public string ChildrenNames { get; set; }
        public List<PresentInfo> Presents { get; set; }
        public DateTime Datetime { get; set; }
        public string Comments { get; set; }
    }
}

File: SantaFinder.Web/Client/src/app/data-services/view-models/new-order/new-order.ts
import { Present } from './present.view-model';
import { Address } from '../../../shared/models/address.model';
import { Location } from '../../../shared/models/location';

export interface NewOrder {
    address: {
        useDefaultAddress: boolean;
        customAddress: {
            line: Address;
            location: Location;
        }
    };
    childrenNames: string;
    presents: Present[];
    datetime: string;
    comments: string;
}